pkg/didmethod/httpbinding/resolver.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package httpbinding
+
+import (
+	"crypto/tls"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"time"
+
+	errors "golang.org/x/xerrors"
+)
+
+// HTTPClientOpts holds options for the HTTP transport implementation of DIDResolver
+// it has a required http.Client instance and optional basePath which is appended to the remove DID Resolver URL
+type HTTPClientOpts struct {
+	client   *http.Client
+	basePath string
+}
+
+// HTTPOptSetter is an outbound HTTP transport option
+type HTTPOptSetter func(opts *HTTPClientOpts)
+
+// WithHTTPClientInstance option is for creating an Outbound HTTP transport using an http.Client instance
+func WithHTTPClientInstance(client *http.Client) HTTPOptSetter {
+	return func(opts *HTTPClientOpts) {
+		opts.client = client
+	}
+}
+
+// WithHTTPClientTimeout option is for creating an Outbound HTTP transport using a client timeout value
+func WithHTTPClientTimeout(timeout time.Duration) HTTPOptSetter {
+	return func(opts *HTTPClientOpts) {
+		opts.client.Timeout = timeout
+	}
+}
+
+// WithHTTPClientTLSConfig option is for creating an Outbound HTTP transport using a tls.Config instance
+func WithHTTPClientTLSConfig(tlsConfig *tls.Config) HTTPOptSetter {
+	return func(opts *HTTPClientOpts) {
+		opts.client = &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: tlsConfig,
+			},
+		}
+	}
+}
+
+// WithHTTPClientBasePath options is for setting optional base path, for example "document"
+func WithHTTPClientBasePath(basePath string) HTTPOptSetter {
+	return func(opts *HTTPClientOpts) {
+		opts.basePath = basePath
+	}
+}
+
+// resolveDid makes DID resolution via HTTP
+func (res *DIDResolver) resolveDid(url string) ([]byte, error) {
+	resp, err := res.client.Get(url)
+	if err != nil {
+		return nil, errors.Errorf("HTTP Get request failed: %w", err)
+	}
+
+	// TODO support for service endpoint URL resolution
+	if containsDidDocument(resp) {
+		var gotBody []byte
+		gotBody, err = ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return nil, errors.Errorf("Failed to read response body: %w", err)
+		}
+
+		err = resp.Body.Close()
```

Comment:
body will not close if err return from ReadAll

Reply:
Agree, good catch!